### PR TITLE
3.11.4

### PR DESCRIPTION
**Pull request by TiceCosmos**
- Native build like 3.12.2
- Cross compilation add protoc native-build

### CODE REVIEW

File: meson.build [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+  protoc_gen = generator(protoc_native,
+    output: ['@BASENAME@.pb.cc', '@BASENAME@.pb.h'],
+    arguments: ['@EXTRA_ARGS@', '--proto_path=' + meson.current_source_dir()/'src', '--cpp_out=@BUILD_DIR@', '@INPUT@']
+  )
+else
+  protoc_gen = generator(protoc,
+    output: ['@BASENAME@.pb.cc', '@BASENAME@.pb.h'],
+    arguments: ['@EXTRA_ARGS@', '--proto_path=' + meson.current_source_dir()/'src', '--cpp_out=@BUILD_DIR@', '@INPUT@']
+  )
+endif
```
Comment (eli-schwartz) [inline meson.build]: Better to set the protoc_native variable to the regular executable, if it's a non-cross build.

Then define one generator outside of the if.

**Review by eli-schwartz — CHANGES_REQUESTED** [review]

**Comment by eli-schwartz** [comment]
https://github.com/mesonbuild/meson/issues/1531 is relevant.

**Comment by eli-schwartz** [comment]
Sorry for the lateness of the reply.

Any particular reason to add an older version when a wrap for 3.12.2 exists already? Why not add protoc-native and protoc_gen to that instead?

**Comment by TiceCosmos** [comment]
> Sorry for the lateness of the reply.
> 
> Any particular reason to add an older version when a wrap for 3.12.2 exists already? Why not add protoc-native and protoc_gen to that instead?

The only difference is version. It's ok to use 3.12.2. I will to add these features to 3.12.2.

**Pull request closed without merging**